### PR TITLE
Unpack freenode changes and add an option for maxHttpSockets

### PR DESCRIPTION
**Pull request by turt2live**
Presence is disabled at the bridge lib layer, and therefore we don't need to comment it out. Similarly, the maxSockets option should be an option.

### CODE REVIEW

**Comment by ara4n** [comment]
this looks plausible to me.

**Comment by ara4n** [comment]
(i assume the various nullguard fixes have made their way in somehow already)

**Comment by turt2live** [comment]
Yea, previous PRs: #674, #677, #679

File: config.sample.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ homeserver:
   # is the "domain name" part of the HS URL.
   domain: "localhost"
 
-  # Should presence be enabledfor matrix clients on this bridge. If disabled on the 
+  # Should presence be enabledfor matrix clients on this bridge. If disabled on the
```
Comment (Half-Shot) [inline config.sample.yaml]: enabledfor

Comment (turt2live) [reply config.sample.yaml]: Historical spelling errors are the worst :(

File: config.sample.yaml [hunk 2]
```diff
@@ -446,3 +446,12 @@ ircService:
     # This is used to stem the flow of requests in case of a mass quit/leave, which might
     # slow down the homeserver.
     leaveConcurrency: 10
+
+# Options here are generally only applicable to large-scale bridges and may have
+# consequences greater than other options in this configuration file.
+advanced:
```
[the remark this answers is not in the document]
Comment (turt2live) [reply config.sample.yaml]: It's not accountability, it's support load. `homeserver` is a more inviting section, and this is not an inviting option. 

Can we at least settle for having it be `homeserver.advanced.maxHttpSockets`? I'm not too fond of having to fight the linter with that long of a name, but I also really don't want to lose the value of saying something is "advanced".

[the remark this answers is not in the document]
Comment (Half-Shot) [reply config.sample.yaml]: I just really don't agree with calling anything advanced. It has no meaning and is a mixed bag of stuff someone decided the boobs couldn't play with. I'd settle for `httpOptions`

[the remark this answers is not in the document]
Comment (Half-Shot) [reply config.sample.yaml]: I just really don't agree with calling anything advanced. It has no meaning and is a mixed bag of stuff someone decided the boobs couldn't play with. I'd settle for `httpOptions`

File: lib/bridge/IrcHandler.js [hunk 3]
```diff
@@ -637,16 +637,6 @@ IrcHandler.prototype.onJoin = Promise.coroutine(function*(req, server, joiningUs
         return Promise.all([
             joinRetry(0),
             intent.setPresence("online")
-/*
```
[the remark this answers is not in the document]
Comment (ara4n) [reply lib/bridge/IrcHandler.js]: it got sucked in via my merge hell

**Comment by ara4n** [comment]
i'm going to merge this so we can deploy it on rizon & moznet - you can rename the config option later if you insist.

**Pull request merged**